src/orion/core/worker/primary_algo.py

```diff
@@ -1,356 +1,109 @@
+    ...
 
 
+@overload
 def create_algo(
-    algo_type: type[AlgoType],
+    algo_type: Callable[Concatenate[Space, P], WarmStarteableAlgo],
     space: Space,
+    knowledge_base: KnowledgeBase,
+    *algo_args,
     **algo_kwargs,
-) -> SpaceTransformAlgoWrapper[AlgoType]:
-    """Creates an algorithm of the given type, taking care of transforming the space if needed."""
-    original_space = space
-    from orion.core.worker.transformer import build_required_space
+) -> InsistSuggest[SpaceTransform[WarmStarteableAlgo]]:
+    ...
 
-    # TODO: We could perhaps eventually *not* wrap the algorithm if it doesn't require any
-    # transformations. For now we just always wrap it.
-    transformed_space = build_required_space(
-        space,
-        type_requirement=algo_type.requires_type,
-        shape_requirement=algo_type.requires_shape,
-        dist_requirement=algo_type.requires_dist,
-    )
-    algorithm = algo_type(transformed_space, **algo_kwargs)
-    wrapped_algo = SpaceTransformAlgoWrapper(algorithm=algorithm, space=original_space)
-    return wrapped_algo
 
+@overload
+def create_algo(
+    algo_type: Callable[Concatenate[Space, P], AlgoType],
+    space: Space,
+    knowledge_base: None = None,
+    *algo_args: P.args,
+    **algo_kwargs: P.kwargs,
+) -> InsistSuggest[SpaceTransform[AlgoType]]:
+    ...
 
-# pylint: disable=too-many-public-methods
-class SpaceTransformAlgoWrapper(BaseAlgorithm, Generic[AlgoType]):
-    """Perform checks on points and transformations. Wrap the primary algorithm.
 
-    1. Checks requirements on the parameter space from algorithms and create the
-    appropriate transformations. Apply transformations before and after methods
-    of the primary algorithm.
-    2. Checks whether incoming and outcoming points are compliant with a space.
+def create_algo(
+    algo_type: Callable[..., AlgoType],
+    space: Space,
+    knowledge_base: KnowledgeBase | None = None,
+    *algo_args: Any,
+    **algo_kwargs: Any,
+) -> InsistSuggest[SpaceTransform[AlgoType]] | MultiTaskWrapper[
+    InsistSuggest[SpaceTransform[AlgoType]]
+]:
+    """Adds different wrappers on top of an algorithm of type `algo_type` before it gets used.
+
+    These wrappers are used to:
+    - apply the transformations required for the algorithm to be applied to the given search space;
+    - check whether the algorithm is warmstarteable and wrap it in a `MultiTaskWrapper` if it isnt;
+    - Make sure that calls to returned algo's `suggest` method returns a Trial, by trying a few times.
 
     Parameters
     ----------
-    algorithm: instance of `BaseAlgorithm`
-        Algorithm to be wrapped.
-    space : `orion.algo.space.Space`
-       The original definition of a problem's parameters space.
-    algorithm_config : dict
-       Configuration for the algorithm.
-
+    algo_type : Callable[..., AlgoType]
+        Type of algorithm to create and wrap.
+    space : Space
+        _description_
```

Comment:
placeholder?

Reply:
Fixed in 3ae6acb5b196cb4a3d373828ba740bf3728b0be4